### PR TITLE
fix(storage): weird caching behaviour with `Option<T>` & replace Mutex with `parking_lot`

### PR DESCRIPTION
**Pull request by Zk2u**
## Description

This PR addresses some weird caching behaviour I found while attempting a rewrite of the CacheTable's inner structure. TLDR is that instead of caching V, we cache Option<V>. That means if it returns None, it will keep returning None until that key is either purged or expired. I'm not sure if this could lead to other issues, but this seemed wrong to me. So this fixes that. 

Also, `parking_lot`'s `Mutex` is so much faster than tokio's or std's, especially considering microcontention cases like this. It has other advantages too, and is just a better Mutex all around. So I'm switching the CacheTable to use that too.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactor
- [x] New or updated tests
- [ ] Dependency Update

## Notes to Reviewers

This is quite a small PR.

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code where necessary.
- [x] I have updated the documentation if needed.
- [x] My changes do not introduce new warnings.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [x] New and existing tests pass with my changes.

## Related Issues

- STR-704
- STR-705

### CODE REVIEW

File: crates/storage/src/cache.rs [hunk 1]
```diff
@@ -20,17 +20,17 @@ pub enum SlotState<T> {
     Ready(T),
 
     /// A database fetch is happening in the background and it will be updated.
-    Pending(broadcast::Receiver<T>),
+    Pending(broadcast::Receiver<Option<T>>),
```
Comment (storopoli) [inline crates/storage/src/cache.rs:23]: I don't get it. Aren't we replacing the `Option<T>` in this PR?
Or are we caching a `T` and maybe we don't have it cached and we return `None`?

Comment (storopoli) [reply crates/storage/src/cache.rs:23]: Oh, makes sense.

**Comment by delbonis** [comment]
> That means if it returns None, it will keep returning None until that key is either purged or expired. 

This is the correct behavior.

**Comment by delbonis** [comment]
Right?  Is there something wrong with my reasoning?

**Comment by delbonis** [comment]
There aren't stale `None`s though, since we (are supposed to) purge the existing cache entries when we write new values.  Adjusting the managers to make it writethrough in the meantime isn't a bad idea.

What I would like in some future version is a new type of cache that's context-aware and meant for storing things that act like `Option<V>` that separately tracks `None`s and doesn't count them against the LRU limit (maybe using a separate LRU limit that's higher instead?).

**Comment by delbonis** [comment]
Also are you *sure* it makes sense to use parking_lot here?  I know we don't hold the lock for lock but this *could* result in a tokio runtime thread being blocked which would be worse.

**Pull request closed without merging**